contracts/DecentHats.sol

```diff
@@ -0,0 +1,485 @@
+                _hat.imageURI
+            );
+    }
+
+    function _createAccount(
+        IERC6551Registry _registry,
+        address _hatsAccountImplementation,
+        address protocolAddress,
+        uint256 hatId
+    ) internal returns (address) {
+        return
+            _registry.createAccount(
+                _hatsAccountImplementation,
+                SALT,
+                block.chainid,
+                protocolAddress,
+                hatId
+            );
+    }
+
+    function _createTopHatAndAccount(
+        IHats _hatsProtocol,
+        string memory _topHatDetails,
+        string memory _topHatImageURI,
+        IERC6551Registry _registry,
+        address _hatsAccountImplementation
+    ) internal returns (uint256 topHatId, address topHatAccount) {
+        topHatId = _hatsProtocol.mintTopHat(
+            address(this),
+            _topHatDetails,
+            _topHatImageURI
+        );
+
+        topHatAccount = _createAccount(
+            _registry,
+            _hatsAccountImplementation,
+            address(_hatsProtocol),
+            topHatId
+        );
+    }
+
+    function _createHatAndAccountAndMintAndStreams(
+        IHats hatsProtocol,
+        IERC6551Registry registry,
+        address topHatAccount,
+        address hatsAccountImplementation,
+        uint256 adminHatId,
+        Hat calldata hat
+    ) internal returns (uint256 hatId, address accountAddress) {
+        hatId = _createHat(
+            hatsProtocol,
+            adminHatId,
+            hat,
+            topHatAccount,
+            topHatAccount
+        );
+        accountAddress = _createAccount(
+            registry,
+            hatsAccountImplementation,
+            address(hatsProtocol),
+            hatId
+        );
+
+        hatsProtocol.mintHat(hatId, hat.wearer);
+
+        for (uint256 i = 0; i < hat.sablierParams.length; ) {
+            _createSablierStream(hat.sablierParams[i], accountAddress);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _createTermedHatAndAccountAndMintAndStreams(
+        IHats hatsProtocol,
+        address topHatAccount,
+        address eligibilityAddress,
+        uint256 adminHatId,
+        Hat calldata hat
+    ) internal {
+        require(
+            hat.termedParams.length == 1,
+            "DecentHats: termedParams length must be 1"
+        );
+        require(
+            hat.termedParams[0].nominatedWearers.length == 1,
+            "DecentHats: nominatedWearers length must be 1"
+        );
+        uint256 hatId = _createHat(
+            hatsProtocol,
+            adminHatId,
+            hat,
+            topHatAccount,
+            eligibilityAddress
+        );
+
+        IHatsElectionEligibility(eligibilityAddress).elect(
+            hat.termedParams[0].termEndDateTs,
+            hat.termedParams[0].nominatedWearers
+        );
+
+        hatsProtocol.mintHat(hatId, hat.termedParams[0].nominatedWearers[0]);
```

Comment:
Definitely open to pushback on this suggestion, but, what if we called `batchMintHat` here and passed in the entire `nominatedWearers` array? Leave it to the frontend to just make sure we're only passing in one of them. That might allow us to continue using this same contract if we do ever decide to open up "role members" to > 1 at a time.

Reply:
Edit: nevermind, this would cause a discrepancy in how we're setting up streams then, cause we are hardcoding the [0] there. And it's beyond us to decide how streams should happen if there are multiple nominated wearers.

So now maybe I'm thinking, can we just turn `nominatedWearers` into a single `address`? We'd need to put that address into an array when calling eligibility.elect, but that's fine. I kind of like that better, making the contract's input params match our business logic.

Reply:
addressed this